src/mesh/GlobalData.cpp

```diff
@@ -0,0 +1,92 @@
+#include "GlobalData.hpp"
+#include <algorithm>
+#include <utility>
+
+#include "SharedPointer.hpp"
+#include "precice/types.hpp"
+#include "utils/assertion.hpp"
+
+namespace precice::mesh {
+
+GlobalData::GlobalData()
+    : _name(""),
+      _id(-1),
+      _dimensions(0),
+      _spatialDimensions(-1)
+{
+  PRECICE_ASSERT(false);
+}
+
+GlobalData::GlobalData(
+    std::string name,
+    DataID      id,
+    int         dimensions,
+    int         spatialDimensions)
+    : _values(),
+      _name(std::move(name)),
+      _id(id),
+      _dimensions(dimensions),
+      _spatialDimensions(spatialDimensions)
+{
+  PRECICE_ASSERT(dimensions > 0, dimensions);
+}
+
+Eigen::VectorXd &GlobalData::values()
+{
+  return _values;
+}
+
+const Eigen::VectorXd &GlobalData::values() const
+{
+  return _values;
+}
+
+// Eigen::MatrixXd &GlobalData::gradientValues()
+// {
+//   return _gradientValues;
+// }
+
+// const Eigen::MatrixXd &GlobalData::gradientValues() const
+// {
+//   return _gradientValues;
+// }
```

Comment:
We will not need gradients for global data

Reply:
yes, this is just commented out copypasta left in for now. I plan to remove it before opening the review.